google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -239,9 +256,9 @@ public CloudSdk createSdk(
 
 
   @Override
-  public Path stageCredentials(String googleUserName) {
-    Path credentials = doStageCredentials(googleUserName);
-    if (credentials != null) {
+  public Optional<Path> stageCredentials(String googleUserName) {
```

Comment:
this code seems to be repeating itself (not introduced by your change). 

I've sent a pr that refactors this underlying code for this. LMKWYT
https://github.com/GoogleCloudPlatform/google-cloud-intellij/pull/1247